docs/refman/loaderyaml.py

```diff
@@ -28,80 +28,169 @@
+    })
+
+    s_object = Map({
+        **d_named_object, **d_feture_check,
+        'long_name': Str(),
+        Optional('extends', default=''): Str(),
+        Optional('notes', default=[]): OrValidator(Seq(Str()), EmptyList()),
+        Optional('warnings', default=[]): OrValidator(Seq(Str()), EmptyList()),
+        Optional('example', default=''): Str(),
+        Optional('methods'): Seq(s_function),
+        Optional('is_container', default=False): Bool()
+    })
+
+class FastTemplate(Template):
+    d_feature_check: T.Dict[str, T.Any] = {
+        'since': '',
+        'deprecated': '',
+    }
+
+    s_posarg = {
+        **d_feature_check,
+        'default': '',
+    }
+
+    s_varargs: T.Dict[str, T.Any] = {
+        **d_feature_check,
+        'min_varargs': -1,
+        'max_varargs': -1,
+    }
+
+    s_kwarg = {
+        **d_feature_check,
+        'required': False,
+        'default': '',
+    }
+
+    s_function = {
+        **d_feature_check,
+        'notes': [],
+        'warnings': [],
+        'example': '',
+        'posargs': {},
+        'optargs': {},
+        'varargs': None,
+        'posargs_inherit': '',
+        'optargs_inherit': '',
+        'varargs_inherit': '',
+        'kwargs': {},
+        'kwargs_inherit': [],
+    }
+
+    s_object = {
+        **d_feature_check,
+        'extends': '',
+        'notes': [],
+        'warnings': [],
+        'example': '',
+        'methods': [],
+        'is_container': False,
+    }
+
 
 class LoaderYAML(LoaderBase):
-    def __init__(self, yaml_dir: Path) -> None:
+    def __init__(self, yaml_dir: Path, strict: bool=True) -> None:
         super().__init__()
         self.yaml_dir = yaml_dir
         self.func_dir = self.yaml_dir / 'functions'
         self.elem_dir = self.yaml_dir / 'elementary'
         self.objs_dir = self.yaml_dir / 'objects'
         self.builtin_dir = self.yaml_dir / 'builtins'
         self.modules_dir = self.yaml_dir / 'modules'
+        self.strict = strict
+
+        template: Template
+        if self.strict:
+            import strictyaml
+            def loader(file: str, template: T.Union[T.Dict, strictyaml.validators.Validator], label: str) -> T.Dict:
+                r: T.Dict = strictyaml.load(file, template, label=label).data
+                return r
 
-    def _process_function_base(self, raw: T.OrderedDict, obj: T.Optional[Object] = None) -> Function:
+            self._load = loader
+            template = StrictTemplate()
+        else:
+            import strictyaml
+            import yaml
+            from yaml import CLoader
+            def loader(file: str, template: T.Union[T.Dict, strictyaml.validators.Validator], label: str) -> T.Dict:
+                return {**template, **yaml.load(file, Loader=CLoader)}
+
+            self._load = loader
+            template = FastTemplate()
+
+        self.template = template
+
+    def _fix_default(self, v: T.Dict) -> None:
```

Comment:
I am not sure we should even attempt to fix this. For instance, this wouldn't catch `default: ON`, `default: NO`, `default: TRUE`, etc.

Reply:
This fix is needed to make the generated man pages from both yaml loaders identical.  It is the only case where yaml type coercion was causing issues, and mostly that was because I used `if arg.default:` to determine whether an argument had a default, which caused problems for `False` and `0`.

Reply:
My point is that we shouldn't even attempt to achieve parity between the two loaders. Instead clearly state that only the `strictyaml` loader is meant for production and the `fastyaml` loader is for development only.

If we try to achieve parity between the two loaders with this fix we are implicitly saying that both can be used and also implicitly commit to fixing issues as they come up (which will be hard and result in confusion if we don't have CI to test it, which again is annoying).

IMHO this is the wrong approach here. I would rather provide both loaders and clearly state that one is for production and the other one can be used for development since it is faster but does **not** always produce the correct results.

Reply:
In that case, how about improving the performance by throwing more threads / [processes](https://docs.python.org/3.8/library/multiprocessing.html) at the Problem?

Of course this will never be as fast as `fastyaml`, but should still provide a major speed improvement.

> and I am perfectly happy to maintain it in a fork

Let's hope it doesn't come to that. We should be able to come to a compromise here.

Reply:
Sorry, I didn't mean to suggest a full-on fork.  Rather what I meant was to maintain a small patch that replaced the yaml loader only.  I don't think it would be necessarily a bad thing, just more work for me.  And, if I ever get tired of it, I can always just go back to the `strictyaml`

As far as trying to use threading to improve strictyaml, I'm not sure I would be willing to go that far.